pkg/internal/cluster/loadbalancer/config.go

```diff
@@ -33,18 +33,21 @@ type ConfigData struct {
 // DefaultConfigTemplate is the loadbalancer config template
 const DefaultConfigTemplate = `# generated by kind
 global
+  stats socket /var/run/haproxy.sock mode 600 level admin
+  stats timeout 2m
   log /dev/log local0
   log /dev/log local1 notice
   daemon
 
 defaults
   log global
   mode tcp
+  option tcpka
```

Comment:
I'm still seeing the same behavior with `clitcpka` and `srvtcpka` (equivalent to `tcpka`), without tweaking timeouts.

Reply:
the `tcpka` avoids that the intermediate systems closes the connection (in our case iptables since we are doing docker port forwarding).
The `kubectl` commands will always die after the inactivity timeout we configure in haproxy if there is no packet between the server and the client.
To avoid this, kubectl should keep the connection open

Reply:
🤔 seems that `TCP keep alive doesn't kick in until the connection has been idle for 2 hours.` by default